src/sentry/api/bases/organization.py

```diff
@@ -385,14 +386,24 @@ def _filter_projects_by_permissions(
             if force_global_perms:
                 span.set_tag("mode", "force_global_perms")
             else:
-                if (
-                    user
-                    and is_active_superuser(request)  # superuser should fetch all projects
-                    or not filter_by_membership  # explicitly requested projects
-                    or include_all_accessible  # requested $all projects
-                ):
+                active_superuser = user and is_active_superuser(request)
+                # Superuser should fetch all projects.
+                # Also check if explicitly requesting all projects with include_all_accessible
+                if active_superuser or include_all_accessible:
                     span.set_tag("mode", "has_project_access")
                     func = request.access.has_project_access
+                # Check if explicitly requesting specific projects
+                elif not filter_by_membership:
+                    if is_active_staff(request):
```

Comment:
The staff check is nested into the `not filter_by_membership` check because the outer check is always true for staff's use case.
Also, the superuser and all-project check above will also fail for _admin's use case (except if you have both superuser and staff cookie, in which case you'll hit superuser which is fine b/c you would've passed with staff anyways).

Reply:
What is the expected behavior when `filter_by_membership` is true? show nothing because staff is not the member?

Reply:
In that case assuming all the other checks above failed as well, we default to filtering by `has_project_membership`